### PR TITLE
Add KB article explaining default rule modes and how to change them

### PR DESCRIPTION
**Pull request by armchairlinguist**
We get a lot of questions about rules unexpectedly appearing in the wrong mode. We have the instructions for fixing it, but not the connection between the problem and the solution - perfect for a KB!

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team

### CODE REVIEW

File: docs/kb/rules/ruleset-default-mode.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+description: Adjust the default mode for a whole ruleset.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:2]:
```suggestion
description: Change the default mode for a ruleset.
```

File: docs/kb/rules/ruleset-default-mode.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+tags:
+  - Rules
+  - Semgrep Registry
+  - Semgrep Code
+append_help_link: true
+---
+
+# Why do new rules keep appearing in Comment or Block mode?
+
+Semgrep AppSec Platform [Policies](/docs/semgrep-code/policies) can contain both individual rules and **rulesets**, which are curated groups of rules recommended for particular purposes. All organizations in the AppSec Platform start with two rulesets: the `default` ruleset and the `comment` ruleset. These rulesets are curated to be good starter packs for security teams to review, and for developers to see comments about, respectively.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:12]:
```suggestion
Semgrep AppSec Platform [Policies](/docs/semgrep-code/policies) can contain both individual rules and **rulesets**, which are curated groups of rules recommended for particular purposes. All organizations start with two rulesets: the `default` ruleset, which is a good starter pack for security teams, and the `comment` ruleset, which is a good starter pack for developers.
```

I was a bit confused about what "These rulesets are curated to be good starter packs for security teams to review, and for developers to see comments about, respectively" meant  -- do you mean the `default` ruleset is good for security teams to start reviewing findings?

File: docs/kb/rules/ruleset-default-mode.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+As Semgrep adds new rules to improve coverage, some of these rules are also added to rulesets. If you have a ruleset added to your organization's policies, any new rules will be added to the ruleset automatically, and become part of your policies also.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:14]:
```suggestion
As Semgrep adds new rules to improve coverage, some of these rules are also added to rulesets. If you add a ruleset to your organization's policies, any new rules added to the ruleset automatically become a part of your policies as well.
```

File: docs/kb/rules/ruleset-default-mode.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+The `default` and `comment` rulesets are initially added in Monitor mode, which is primarily intended for security teams to review. You can also [add new rulesets to your policies](/docs/semgrep-code/policies#add-rulesets-to-your-policies-from-the-registry) from the Semgrep Registry.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:16]:
```suggestion
The `default` and `comment` rulesets are initially added in **Monitor** mode, where the findings generated by the rules are primarily intended for security teams to review. You can also [add new rulesets to your policies](/docs/semgrep-code/policies#add-rulesets-to-your-policies-from-the-registry) from the Semgrep Registry.
```
I don't know if I captured the gist of what you're saying, but I think I'm trying to clarify whether the security team is reviewing the rules themselves or the findings they generate

File: docs/kb/rules/ruleset-default-mode.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+When you add a ruleset through the registry, you can add it in any policy mode: Monitor, Comment, or Block. The mode you choose will determine the mode for future rules that are added to that ruleset.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:18]:
```suggestion
When you add a ruleset through the registry, you can add it in any policy mode: **Monitor**, **Comment**, or **Block**. The mode you choose will determine the mode for future rules that are added to that ruleset.
```

File: docs/kb/rules/ruleset-default-mode.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+Even if you later move some or all rules from a ruleset to a different mode, the default mode for the ruleset remains the same as it was when it was originally added, and it might be somewhat surprising when new rules are added in the original mode.
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:20]:
```suggestion
Even if you later change some or all rules from a ruleset to a different mode, the default mode for the ruleset does not change. Therefore, when you add new rules to the ruleset, they are added in the original mode.
```

File: docs/kb/rules/ruleset-default-mode.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+## Changing the default mode for a ruleset
```
Comment (khorne3) [inline docs/kb/rules/ruleset-default-mode.md:22]:
```suggestion
## Change the default mode for a ruleset
```

**Comment by relsqui** [comment]
I agree with Katie's comments and don't feel a need to give you two pending reviews to check off for the same changes. :)

File: docs/kb/rules/ruleset-default-mode.md (hunk 2 above)
Comment (armchairlinguist) [reply docs/kb/rules/ruleset-default-mode.md:12]: yeah, your version is better - that was some bad writing on my part, sorry!

**Comment by armchairlinguist** [comment]
Accepted all the suggestions, fresh review welcome :)

**Review by khorne3 — APPROVED** [review]

**Pull request merged**